fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    fetch('http://localhost:3000/api/todos')
+      .then(r => r.json())
+      .then(todos => this.setState({ todos }))
+      .catch(err => console.error(err));
+  }
+
+  addTodo(newTodo) {
+    fetch('http://localhost:3000/api/todos', {
+      method: 'POST',
+      body: JSON.stringify(newTodo),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    })
+      .then(r => r.json())
+      .then(r => {
+        const todosArray = this.state.todos;
+        const newArray = todosArray.concat(r);
+        this.setState({ todos: newArray });
+      }
+      );
+
+  }
+
+  toggleCompleted(todoId) {
+    const todos = this.state.todos;
```

Comment:
Be sure to make a shallow copy of this before you assign the new todo object to the target index. A new variable is not a copy, because objects (including arrays and functions) are pass-by-reference